### ISSUE TITLE
Boot is taking quite some time (approx 3 mins)

### ISSUE THREAD

**Issue by aitsamahad** [jsassu20/Lenovo-ThinkPad-T450-macOS-Catalina#45]
Hi @jsassu20 , first of all i want to thank you for all of the wonderful work that you have already done, you are awesome!. 
Now onto my issue, I don't know if its only my issue or this is normal for ThinkPad T450 Hackintosh.
My System is T450, i5-5300U.
I have followed through your process of hackintosh of Mojave (3.1) -[https://github.com/jsassu20/Lenovo-ThinkPad-T450-macOS-Catalina/releases/tag/3.1] and copied your EFI folder. Everything that supposed to be working is working fine, just boot is taking quite some time. any solution for that? i have another laptop Lenovo Yoga Y10 and i made that into hackintosh too and its overall boot time is less then 30 secs.

**Comment by surrogatecode** [comment]
Hi. 

Have you tried a verbose boot (clover boot settings: -v, I believe) to see at what point it is hanging for a longer time? That might help to find out what is slowing down the T450.  I think you can also log the boot messages and post them here for jsassu20 or another clover-savvy user.

**Comment by aitsamahad** [comment]
Yes i did verbose boot,
The problem i found after Booting Verbose is that its getting stuck for some time on "FIPSPOST_USER ... all test passed", this thing is i did found a forum discussion about that but it was not much detailed but the solution mentioned there was simply to remove TRIM from SSD or Format the SSD to HFS+J instead of APFS. I don't know about FileSystem solution but I don't think removing TRIM will help me because this boot hanging process is been happening even before i enabled TRIM myself and it didn't get any better or worse after enabling it.

After FIPSPOST_USER ... all test passed, it shows "Previous Shutdown cause: 5" and then just boot into the OS immediately.

If you want me to give any other detail or anything, do tell me.
Any help will be appreciated, Thanks

**Comment by surrogatecode** [comment]
disclaimer: I have no idea. 

I wanted to suggest, to look into whatever is appearing in the log-file after FIPSPOST_USER has been passed, because based on your and [other people's](https://www.tonymacx86.com/threads/solved-boot-hangs-at-fipspost_post.247544/) description it seems to get stuck **after** passing the tests. However, if this is just "Previous Shutdown Cause: 5" I am quite uncertain. Afaik, that is a normal shutdown?

If you have a spare SSD, you could try making a mirror of your system on a HFS+ format and see if that boots without issues as people claim.

Good luck finding a solution.

**Comment by aitsamahad** [comment]
@surrogatecode Yes the shutdown is normal as the code itself suggest "5" is supposed to be a normal standard shutdown.
I don't have an spare SSD but i was thinking of doing a clean install if i find a suitable solution, but i have to ask, isn't from Mojave and further installation of OS on HFS+J is not supported on on APFS (I might be wrong, but it is what i heard so far).
Thank you for your time, i really appreciate it, and if you find another post related to this or solution for this then do please let me know. Cheers!

**Comment by surrogatecode** [comment]
Oh, I am sorry. Yes, you are correct, Mojave's standard partition format is APFS. I guess I made this mistake after reading the above-mentioned forum post (for High Sierra). Nonetheless, if you plan on doing a clean install, you could try the method mentioned in that forum post:

> 5. Erase the (...) drive and format it to HFS+.
> 6. Erase (...) drive (again) and format it to APFS.

This sounds lke a hack, rather than a fix. But maybe it will work.

**Comment by aitsamahad** [comment]
@surrogatecode Yes this feels and sound like a hack rather then a fix, but if this solves my problem then i am all for it.
Anyways, ill start removing my important files and folders from my OS and then ill try this fix(hack).
Will update here if that works.
Thanks mate!

**Comment by surrogatecode** [comment]
Did this work out in the end?

**Comment by aitsamahad** [comment]
@surrogatecode sorry for very late reply... and no that fix(hack) didn't work.
